src/springfield/entity.py

```diff
@@ -94,9 +94,11 @@ def flatten(self):
         Get the values as basic Python types
         """
         data = {}
-        for key, val in self.__values__.iteritems():
-            val = self.__fields__[key].flatten(val)
-            data[key] = val
+        for key, field in self.__fields__.iteritems():
```

Comment:
`flatten` and `jsonify` should explicitly not return defaults values. SpringField honors sparse objects and sparse updates. So if a value was no explicitly set, it should not be validated. Defaults are only meant for use when using an Entity object directly (see other comments about default test).

Here's an example illustrates this.

1. You have a database with 4 fields: id, name, enabled, and created
2. You have this entity:

```
class UserEntity(Entity):
    id = fields.UuidField()
    name = fields.StringField()
    enabled = fields.BooleanField(default=True)
    created = fields.DateField(default=utcnow)
```
3. You create user record with `user = UserEntity(id=uuid4(), name = 'Bob Test').flatten()`. Because of defautls `user['enabled'] is True`.
4. If defaults are flattened, the database record looks like `{"id": "AAAA", "name": "Bob Test", "default": True, "created": "2016-04-06T00:00:00Z"}`. This is actually good because our defaults were populated when we create the record.
5. Say the user has left the company so you want to disable him. So you UPDATE with values `UserEntity(enabled=False).flatten()`. Since sparse updates are desired and there is no default for `id` or `name`, those aren't changed. However `created` has a default so it gets updated with a new timestamp but it should never change since you can only be created once.
6. To make it less confusing to other users, you want to change his name to "Bob Test (disabled)" so you UPDATE with `UserEntity(name='Bob Test (disabled)').flatten()` and defaults strike again. Now `enabled is True` because that's the default and `created` is set to now. 
7. The database record is in an invalid state.

We use sparse updates quite frequently so suddenly having defaults in flatten will cause many problems.

However, as illustrated above, defaults are useful for INSERTs. When a record is first created is the time to evaluate defaults. A simple solution is to do `def flatten(include_defaults=False)`. This way, by defaults `flatten` will only include set values and you can use `include_defaults=True` only for the INSERT case.

While I think this is a good balance, the problem gets a little more complicated which I'll illustrate by talking about `flatten`'s sibling `jsonify`.